freud/box.pyx

```diff
@@ -228,8 +228,7 @@ cdef class Box:
                 "Invalid dimensions for fractions given to makeCoordinates. "
                 "Valid input is an array of shape (3,) or (N,3).")
 
-        fractions = freud.common.convert_array(
-            fractions, fractions.ndim, dtype=np.float32, contiguous=True)
+        fractions = freud.common.convert_array(fractions, fractions.ndim)
```

Comment:
There are a couple cases like this one (and `face_orientations` in the PMFT module) where we're not actually doing anything with the number of dimensions. I'd like to see `dimensions=None` by default, and remove this parameter from calls to `convert_array` where we don't enforce a specific array dimensionality. This would require that we change the logic in `convert_array` to `if dimensions is not None and array.ndim != dimensions:`.